common.js

```diff
@@ -2545,6 +2545,218 @@ const allSoloCars = {
       '2004': ['ep'],
     },
   },
+  'Saab': {
+    '9-2X': {
+      '2005': ['sth'],
+      '2006': ['sth']
+    },
+    '9-2X Aero (2.0L Turbo)': {
+      '2005': ['ds'],
+      '2006': ['ds']
+    },
+    '9-2X Linear (2.5L)': {
+      'all': ['gs']
+    },
+    '9-3 (non-Viggen)': {
+      '1998': ['stx'],
+      '1999': ['stx'],
+      '2000': ['stx'],
+      '2001': ['stx'],
+      '2002': ['stx'],
+      '2003': ['stx'],
+      '2004': ['stx'],
+      '2005': ['stx'],
+      '2006': ['stx'],
+      '2007': ['stx'],
+      '2008': ['stx'],
+      '2009': ['stx'],
+      '2010': ['stx'],
+      '2011': ['stx'],
+      '2012': ['stx']
+    },
+    '900': {
+      '1979': ['dsp'],
+      '1980': ['dsp'],
+      '1981': ['dsp'],
+      '1982': ['dsp'],
+      '1983': ['dsp'],
+      '1984': ['dsp'],
+      '1985': ['dsp'],
+      '1986': ['dsp'],
+      '1987': ['dsp'],
+      '1988': ['dsp'],
+      '1989': ['dsp'],
+      '1990': ['dsp'],
+      '1991': ['dsp'],
+      '1992': ['dsp'],
+      '1993': ['dsp'],
+      '1994': ['dsp'],
+      '1995': ['dsp'],
+      '1996': ['dsp'],
+      '1997': ['dsp'],
+      '1998': ['dsp']
+    },
+    '900 (non-turbo, FWD)': {
+      '1979': ['ep'],
+      '1980': ['ep'],
+      '1981': ['ep'],
+      '1982': ['ep'],
+      '1983': ['ep'],
+      '1984': ['ep'],
+      '1985': ['ep'],
+      '1986': ['ep'],
+      '1987': ['ep'],
+      '1988': ['ep'],
+      '1989': ['ep'],
+      '1990': ['ep'],
+      '1991': ['ep'],
+      '1992': ['ep'],
+      '1993': ['ep'],
+      '1994': ['ep']
+    },
+    '900 (V6)': {
+      '1994': ['hs'],
+      '1995': ['hs'],
+      '1996': ['hs'],
+      '1997': ['hs']
+    },
+    '900 SPG Turbo 16v': {
+      '1979': ['fp'],
+      '1980': ['fp'],
+      '1981': ['fp'],
+      '1982': ['fp'],
+      '1983': ['fp'],
+      '1984': ['fp'],
+      '1985': ['fp'],
+      '1986': ['fp'],
+      '1987': ['fp'],
+      '1988': ['fp']
+    },
+    '900 Turbo': {
```

Comment:
You shouldn't have duplicate keys. You have `900 Turbo` twice. Please merge these two, by adding `fp` to all years or by removing this model and adding `fsp` to the model with `all` for the years.